Modified table class name for non-cached UserAgent

Jordan9675 opened the pull request:
It seems like the class attribute associated to the table on https://www.w3schools.com/browsers/default.asp has changed thus leading to `IndexError` when trying to initialize `UserAgent` with `cache=False`

Jordan9675 committed:
Modified table class name for non-cached UserAgent

thebetauser commented:
Confirmed this fixes it the issues, need one of the admins to merge this request to main branch

toh995 commented:
Thank you for catching this, hopefully it gets merged soon :pray:

pmartynov commented:
@hellysmile who is managing this repo? I need this fix to get merged.

peterschutt commented:
@RapidEU fork it and do it yourself.. no one is obligated to do anything for you.

maxkrivich reviewed (commented):
https://github.com/hellysmile/fake-useragent/blob/460686d0c729c2b22172dc75760408c87b7d6618/fake_useragent/settings.py#L7

Do not forget to update the version.

Jordan9675 commented:
> https://github.com/hellysmile/fake-useragent/blob/460686d0c729c2b22172dc75760408c87b7d6618/fake_useragent/settings.py#L7
> 
> Do not forget to update the version.

Thanks for pointing this out, I changed it and added it within the commit

MattDMo commented:
bump....

Treeless commented:
Please merge this Admins!!!

arunkannawadi commented:
The error you see is not an actual error but a traceback from an exception that _is_ handled internally. You should be able to ignore the apparent error and continue running the code. However, if you'd like to suppress the annoying/confusing message , you can temporarily redirect your output into the void using this nice little trick:
https://thesmithfam.org/blog/2012/10/25/temporarily-suppress-console-output-in-python/

lucawen commented:
this repo is dead

MattDMo commented:
@arunkannawadi No, this is an actual error that prevents the module from generating a list of browser User-Agent strings. Hiding the error isn't going to make it go away, you need to update the code as detailed in this PR.

melroy89 commented:
Maybe we should fork it and create a new pip package.  I see that @hellysmile is from Ukraine. 

I will do it: https://github.com/danger89/fake-useragent

MattDMo commented:
@hellysmile 's last contribution to Github was on 2022-04-14, and there has been no activity since. During the previous year, he was fairly consistent in his activity, with the longest lag being about a month in Nov-Dec 2021. How long do we wait before [petitioning PyPI](https://peps.python.org/pep-0541/#how-to-request-a-name-transfer) to change the `fake_useragent` project to use @danger89 's fork? Is everyone okay with that fork? I am.

I just sent an email to hellysmile@gmail.com, asking Victor if he intends to come back to this repo, or if he'd mind if @danger89 take over in some manner. I suggest waiting a few weeks or a month for a reply, and if I don't hear from him we can put together the PEP 541 info.

melroy89 commented:
I'm fine with that as well.

sgerodes commented:
I am ok with that too.

MattDMo commented:
Well, it's been over 2 months, and I haven't heard a thing, so it looks like @hellysmile is either unwilling or unable to respond, for whatever reason. @danger89, would you like to start the petitioning process with PyPI? Feel free to let me know if you need help. I don't have a ton of time to devote to this, but I'd like to see it through to completion.

melroy89 commented:
> Well, it's been over 2 months, and I haven't heard a thing, so it looks like @hellysmile is either unwilling or unable to respond, for whatever reason. @danger89, would you like to start the petitioning process with PyPI? Feel free to let me know if you need help. I don't have a ton of time to devote to this, but I'd like to see it through to completion.

Thanks for your time and trying to reach out to @hellysmile. 

I will look into the PyPi transfer procedure and create an issue report soon at there side. Let's do this!

melroy89 commented:
FYI. Transfer ticket created: https://github.com/pypa/pypi-support/issues/2313

melroy89 commented:
@hellysmile Looks like he is back. I'm now owner of the repo as well as the pypi package. 

I will merge my changes back upstream https://github.com/hellysmile/fake-useragent/pull/136

chrisspen commented:
Still getting this error in 1.4.0. Any chance of getting a new release?

melroy89 commented:
Are you sure you updated the right package and are on V1. 4.0? Since all this table code and cache is gone. We don't scrape data like that anymore. We also have all user agent data locally stored within the package.

```
pip install --upgrade fake-useragent
```

chrisspen commented:
Ah, you're right. I had upgraded, but I was activating the wrong virtual environment. Thanks.

melroy89 commented:
Np.
